addons/bestja_detailed_reports/models.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api, exceptions
+from dateutil import parser
+
+
+class Project(models.Model):
+    _inherit = 'bestja.project'
+
+    detailed_reports = fields.One2many('bestja.detailed_report', inverse_name='project')
+    use_detailed_reports = fields.Boolean(required=True, string="Raporty szacunkowe do zbiórki żywności", default=True)
+
+
+class CommodityGroup (models.Model):
+    _name = 'bestja.commodity_group'
+    _order = 'code'
+    code = fields.Char(string="Kod")
+    name = fields.Char(string="Nazwa")
+
+
+class ReportEntry(models.Model):
+    _name = 'bestja.report_entry'
+    _order = 'commodity'
+
+    detailed_report = fields.Many2one('bestja.detailed_report', ondelete='cascade', required=True)
+    commodity = fields.Many2one('bestja.commodity_group', string="Nazwa", required=True, ondelete='cascade')
+    commodity_code = fields.Char(string='Kod', related="commodity.code")
+    tonnage = fields.Float(required=True, string="tonaż (kg)")
+    interested_organization = fields.Many2one(
+        'organization',
+        string="Zainteresowana organizacja",
+        store=True,  # Needed by graph view
+        related='detailed_report.interested_organization',
+    )
+    organization = fields.Many2one(
+        'organization',
+        string="Organizacja",
+        store=True,
+        related='detailed_report.project.organization',
+    )
+
+    _sql_constraints = [
+        ('report_entries_uniq',
+        'unique("detailed_report", "commodity")',
+        "Dany element można wybrać tylko raz!")
+    ]
+
+
+class DetailedReport(models.Model):
+    _name = 'bestja.detailed_report'
+    _inherit = [
+        'protected_fields.mixin',
+        'ir.needaction_mixin',
+        'message_template.mixin',
+    ]
+    _order = 'write_uid desc'
+    STATES = [
+        ('sent', "wysłany"),
+        ('accepted', "zaakceptowany"),
+        ('draft', "szkic"),
+        ('rejected', "odrzucony"),
+    ]
+
+    project = fields.Many2one(
+        'bestja.project',
+        required=True,
+        string="Projekt",
+        domain=lambda self: [
+            ('parent.use_detailed_reports', '=', True),
+            ('detailed_reports', '=', False),
+            '|',
+            ('manager', '=', self.env.user.id),
```

Comment:
I always use indentation for rules covered by an operator. Maybe it is worth to keep this convention?